### PR TITLE
aws: add support for Calyptia integration

### PR DESCRIPTION
**Pull request by patrick-stephens**
Integrate directly with Calyptia Core for upcoming Fleet management.

Addresses #47.

### CODE REVIEW

File: charts/aws-fluent-bit/templates/clusterrole.yaml [hunk 1]
```diff
@@ -18,3 +19,24 @@ rules:
       - get
       - list
       - watch
+  {{- if and .Values.podSecurityPolicy.create (semverCompare "<=1.25-0" .Capabilities.KubeVersion.GitVersion) }}
```
Comment (vigneshcommits) [inline charts/aws-fluent-bit/templates/clusterrole.yaml:22]: @patrick-stephens 
I think this is not rendering as we expect to. 
I have set the  **.Values.podSecurityPolicy.create** to _true_ and I'm on the k8s version 1.25 but i still get to see the podSecurityPolicy created

**Comment by vigneshcommits** [comment]
This works fine. i think it would be better to add a comment for Values.securityContext to set the UID as per the underlying host. what do you think.

File: charts/aws-fluent-bit/templates/clusterrole.yaml (hunk 1 above)
Comment (patrick-stephens) [reply charts/aws-fluent-bit/templates/clusterrole.yaml:22]: Not sure I follow: do you see it on 1.25 as that is expected with `<=`?

**Comment by patrick-stephens** [comment]
> This works fine. i think it would be better to add a comment for Values.securityContext to set the UID as per the underlying host. what do you think.

I think we should separate this into a different issue - what's the underlying cause on EKS for that?

File: charts/aws-fluent-bit/templates/clusterrole.yaml (hunk 1 above)
Comment (vigneshcommits) [reply charts/aws-fluent-bit/templates/clusterrole.yaml:22]: I think PodSecurityPolicies are not allowed in 1.25 version
but i see that the podSecurityPolicy is created.

**Comment by vigneshcommits** [comment]
> > This works fine. i think it would be better to add a comment for Values.securityContext to set the UID as per the underlying host. what do you think.
> 
> I think we should separate this into a different issue - what's the underlying cause on EKS for that?

yes, the EKS nodes i had was on top of Amazon Linux worker nodes.

**Review by vigneshcommits — APPROVED** [review]

**Pull request merged**